### ISSUE TITLE
Gulp watch doesn't work correctly when making a change on scss file

### ISSUE THREAD

**Issue by yanan** [liferay/liferay-themes-sdk#142]
On Ubuntu 18, I'm running into the same issue like https://github.com/liferay/liferay-themes-sdk/issues/102, I can't reopen it so I create a new issue here.

- Portal: liferay-dxp-tomcat-7.1.10.1-sp1-20190110085705206.zip(LPS-86972 and LPS-87487 has been fixed to using install command with single quote)
- node v10.15.1
- npm 6.4.1
- gulp CLI version 2.0.1

Watch works ok on changing a ftl file, but it doesn't work correctly on chaning a scss file.

Steps to reproduce:
1. create a liferay 7.1 theme using "yo liferay-theme" 
2. start portal 71 sp1 tomcat in developer mode
3. execute "gulp watch" and we can see the active state of this project in Gogo shell

> gulp watch
> [13:53:30] Using gulpfile ~/my-theme-test-theme/gulpfile.js
> [13:53:30] Starting 'watch'...
> [13:53:30] Starting 'build'...
> [13:53:30] Starting 'build:clean'...
> [13:53:30] Finished 'watch' after 5.02 ms
> [13:53:30] Finished 'build:clean' after 102 ms
> [13:53:30] Starting 'build:base'...
> [13:53:32] Finished 'build:base' after 1.63 s
> [13:53:32] Starting 'build:src'...
> [13:53:32] Finished 'build:src' after 7.21 ms
> [13:53:32] Starting 'build:web-inf'...
> [13:53:32] Finished 'build:web-inf' after 742 μs
> [13:53:32] Starting 'build:liferay-look-and-feel'...
> [13:53:32] Finished 'build:liferay-look-and-feel' after 16 ms
> [13:53:32] Starting 'build:hook'...
> [13:53:32] Finished 'build:hook' after 8.89 ms
> [13:53:32] Starting 'build:themelets'...
> [13:53:32] Starting 'build:themelet-src'...
> [13:53:32] Finished 'build:themelet-src' after 2.11 ms
> [13:53:32] Starting 'build:themelet-css-inject'...
> [13:53:32] Starting 'build:themelet-js-inject'...
> [13:53:32] gulp-inject Nothing to inject into _custom.scss.
> [13:53:32] gulp-inject Nothing to inject into portal_normal.ftl.
> [13:53:32] Finished 'build:themelet-css-inject' after 51 ms
> [13:53:32] Finished 'build:themelet-js-inject' after 12 ms
> [13:53:32] Finished 'build:themelets' after 54 ms
> [13:53:32] Starting 'build:rename-css-dir'...
> [13:53:32] Finished 'build:rename-css-dir' after 439 μs
> [13:53:32] Starting 'build:compile-css'...
> [13:53:32] Starting 'build:compile-lib-sass'...
> [13:53:33] Finished 'build:compile-lib-sass' after 925 ms
> [13:53:33] Finished 'build:compile-css' after 925 ms
> [13:53:33] Starting 'build:fix-url-functions'...
> [13:53:33] Finished 'build:fix-url-functions' after 45 ms
> [13:53:33] Starting 'build:move-compiled-css'...
> [13:53:33] Finished 'build:move-compiled-css' after 472 ms
> [13:53:33] Starting 'build:remove-old-css-dir'...
> [13:53:33] Finished 'build:remove-old-css-dir' after 27 ms
> [13:53:33] Starting 'build:fix-at-directives'...
> [13:53:33] Finished 'build:fix-at-directives' after 23 ms
> [13:53:33] Starting 'build:r2'...
> [13:53:33] Finished 'build:r2' after 133 ms
> [13:53:33] Starting 'build:war'...
> [13:53:33] Starting 'plugin:version'...
> [13:53:33] Finished 'plugin:version' after 2.22 ms
> [13:53:33] Starting 'plugin:war'...
> [13:53:35] Finished 'plugin:war' after 1.27 s
> [13:53:35] Finished 'build:war' after 1.28 s
> [13:53:35] Finished 'build' after 4.73 s
> [13:53:35] Starting 'watch:clean'...
> [13:53:35] Finished 'watch:clean' after 870 μs
> [13:53:35] Starting 'watch:osgi:clean'...
> [13:53:35] Finished 'watch:osgi:clean' after 14 ms
> [13:53:35] Starting 'watch:setup'...
> [13:53:36] Finished 'watch:setup' after 1.58 s
> [Browsersync] Proxying: http://localhost:8080
> [Browsersync] Access URLs:
>  -------------------------------------
>     Local: http://localhost:9080
>  External: http://192.168.133.129:9080
>  -------------------------------------

> 1 theme for my-theme-test-theme is available for use

4. go to localhost:9080 and apply our theme, it works
5. change src/css/_custom.scss file, for example

> body {
>     background-color:blue;
>  }
6. save scss file chagnes and notice watch detected the changes automatically and _custom.scss also changed in .web_bundle_build/css folder

> [13:58:06] Starting 'deploy:file'...
> [13:58:06] gulp-debug: src/css/_custom.scss
> [13:58:06] gulp-debug: 1 item
> [13:58:06] Finished 'deploy:file' after 18 ms
> [Browsersync] Reloading Browsers...
7. refresh webpage, notice it doesn't catch our chagnes
8. if changing a .ftl file, watch detected changes and refresh browser can see the changes successfully

> [14:10:13] Starting 'deploy:file'...
> [14:10:13] gulp-debug: src/templates/portal_normal.ftl
> [14:10:13] gulp-debug: 1 item
> [14:10:13] Finished 'deploy:file' after 26 ms
> [Browsersync] Reloading Browsers...
9. It seems "deploy:css-files" isn't triggered by changing a scss file.

Can anyone help to take a look? 
Thanks

**Comment by vincentl7** [comment]
Hi team, 
The issue is actually from one of our Enterprise customers, it is currently not a fire but time sensitive. So any help will be greatly appreciated. 
Thanks,

**Comment by jbalsas** [comment]
Hey @yanan, @vincentl7, if you think this is the same as #102, then there's very little we could do from here as that was a Liferay Portal issue that got addressed in a later FP or SP.

Could you please verify if the watch task still fails to work on a more current version of Liferay DXP? Also, could you please verify that you've enabled the developer properties to avoid the Portal aggressive caching behaviour of static files?

Thanks!

**Comment by vincentl7** [comment]
Hi @jbalsas, Noted with thanks! LPP-33004 was originally created for this, and #142 was as a result of JIRA ticket discussion. We will continue using JIRA as main communication channel as suggested then.

**Comment by yanan** [comment]
Hello @jbalsas,
Thanks for the notes.
Tested with latest portal in 7.1.x_private branch, I still run into the same problem. And for 

> could you please verify that you've enabled the developer properties to avoid the Portal aggressive caching behaviour of static files?

I've added "include-and-override=portal-developer.properties" into portal-ext.properties file before starting tomcat. Do you mean to enable this?
I'm going to close this since I've posted some details for current issue into LPP-33004 and assign this ticket to you.Please help take a look when you're available, thanks so much!
